orchestrator/src/test/java/com/scottlogic/deg/orchestrator/cucumber/features/operators/general/EqualTo.feature

```diff
@@ -475,1103 +131,118 @@ Feature: User can specify that a value is equalTo a required value
-    And foo is equal to 9999-12-31T23:59:59.999Z
-    And foo is anything but of type "string"
-    Then the following data should be generated:
-      | foo                      |
-      | null                     |
-      | 9999-12-31T23:59:59.999Z |
+    And foo is equal to "BBB"
+    And foo is matching regex /[a]{3}/
+    Then no data is created
+
+### ofLength ###
 
-  Scenario: 'EqualTo' a string and not 'ofType' decimal is successful
+  Scenario: 'EqualTo' alongside a non-contradicting 'ofLength' constraint should be successful
     Given there is a field foo
-    And foo is equal to "a"
-    And foo is anything but of type "decimal"
+    And foo is equal to "1"
+    And foo is of length 1
     Then the following data should be generated:
       | foo  |
-      | null |
-      | "a"  |
+      | "1"  |
 
-  Scenario: 'EqualTo' a date and not 'ofType' decimal is successful
+  Scenario: 'EqualTo' alongside a contradicting 'ofLength' constraint should emit no data
     Given there is a field foo
-    And foo is equal to 2019-01-01T00:00:00.000Z
-    And foo is anything but of type "decimal"
+    And foo is equal to "1"
+    And foo is of length 2
+    Then no data is created
+
+### Financial code types ###
+
+  Scenario: Equal to a valid ISIN combined with an ISIN constraint should generate the equal to value
+    Given there is a field foo
+    And foo is equal to "GB0002634946"
+    And foo is of type "ISIN"
     Then the following data should be generated:
-      | foo                      |
-      | null                     |
-      | 2019-01-01T00:00:00.000Z |
+      | foo            |
+      | "GB0002634946" |
 
-  Scenario: 'EqualTo' a string and not 'ofType' datetime is successful
+  Scenario: Equal to something that is not a valid ISIN combined with a non-ISIN constraint should generate the equal to value
     Given there is a field foo
     And foo is equal to "a"
-    And foo is anything but of type "datetime"
+    And foo is anything but of type "ISIN"
     Then the following data should be generated:
       | foo  |
-      | null |
       | "a"  |
 
-  Scenario: 'EqualTo' a number and not 'ofType' datetime is successful
+  Scenario: Equal to something that is not a valid ISIN because its check digit is wrong combined with an ISIN constraint should generate no data
     Given there is a field foo
-    And foo is equal to 1
-    And foo is anything but of type "datetime"
-    Then the following data should be generated:
-      | foo  |
-      | null |
-      | 1    |
+    And foo is equal to "GB00026349"
+    And foo is of type "ISIN"
+    Then no data is created
+
+  Scenario: Equal to something that is not a valid ISIN combined with an ISIN constraint should only generate null
+    Given there is a field foo
+    And foo is equal to "aa"
+    And foo is of type "ISIN"
+    Then no data is created
 
-  Scenario: Not 'equalTo' a number and 'ofType' string is successful
+  Scenario: equalTo should interact with sedol
```

Comment:
Not sure about this title

Reply:
Should interact with SEDOL is a bit odd. Why not matching the ISIN format (`Equal to a valid ISIN combined with an ISIN constraint should generate the equal to value`)?

Reply:
Also why no CUSIP

Reply:
& RIC